### PR TITLE
Sticky navigation

### PR DESCRIPTION
**Pull request by bmallred**
Linux
---
 - [x] Chromium
 - [x] Chrome
 - [x] Firefox

Mac
---
 - [x] Chrome
 - [x] Firefox
 - [ ] Safari (does not support `position: sticky`)

Issue: #1520

### CODE REVIEW

File: src/components/Sticky/Sticky.jsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    // Set the state
+    this.setState(future)
+  }
+
+  render () {
+    const styleContainer = {
+      position: this.state.position
+    }
+    const style = {
+      top: this.state.top ? `${this.state.top}${UNIT}` : null
```
Comment (floresj) [inline src/components/Sticky/Sticky.jsx]: Just noticed in my editor that `UNIT` is not defined.

Comment (bmallred) [inline src/components/Sticky/Sticky.jsx]: Ah! Missed that on refactor

**Comment by floresj** [comment]
## Pre-stick
selection_299

## On stick
selection_300

**Comment by floresj** [comment]
Just general question/comments. I think we'll eventually need to support the ability to toggle a css class when it determines something is sticky or not. This will be useful for the header/accordion sticky stuff because we'll need to toggle a class in order to change the styling for those headers and its children once it sticks. Currently, `sticky-container` and `sticky` are always present (not a bad thing).

**Comment by bmallred** [comment]
@floresj just added that to handle the `sticky` value for Safari. It is on the container at the moment.

I'm completely open to using something different for the other stickiness areas. This one seems to be highly customized due to the upper + lower bounds.

**Comment by floresj** [comment]
@Jay-Neighbours-I72 This is functioning fine for me. You have any feedback before I push this through?

**Pull request merged**